examples/conanfile/layout/third_party_libraries/conanfile.py

```diff
@@ -0,0 +1,46 @@
+import os
+from conan import ConanFile
+from conan.tools.cmake import cmake_layout, CMake
+from conan.tools.files import get
+from conan.tools.files import patch
+
+
+class Pkg(ConanFile):
+    name = "hello"
+    version = "1.0"
+    exports_sources = "patches*"
+    generators = "CMakeToolchain"
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+        # if you are declaring your own layout, just declare:
+        # self.folders.source = "src"
+    
+    def source(self):
+        # we are inside a "src" subfolder, as defined by layout
+        # the downloaded soures will be inside the "src" subfolder
+        get(self, "https://github.com/conan-io/libhello/archive/refs/heads/main.zip", 
```

Comment:
Note about this being strongly discouraged-or completely forbidden (depend on head)